src/Tribe/Admin/Notice/Rest_Api.php

```diff
@@ -0,0 +1,122 @@
+<?php
+/**
+ * Shows an admin notice when our REST API endpoints are not available.
+ *
+ * @since
+ *
+ * @package Tribe\Events\Admin\Notice
+ */
+
+namespace Tribe\Events\Admin\Notice;
+
+/**
+ * Class Rest_Api
+ *
+ * Shows an admin notice when our REST API endpoints are not available.
+ *
+ * @since
+ */
+class Rest_Api {
+
+	/**
+	 * Notice Slug on the user options
+	 *
+	 * @since
+	 *
+	 * @var string
+	 */
+	private $slug = 'events-rest-api-notice';
+
+	/**
+	 * Blocked endpoint.
+	 *
+	 * @since
+	 *
+	 * @var string
+	 */
+	private $blocked_endpoint = '';
+
+	/**
+	 * Constructor.
+	 *
+	 * @since
+	 */
+	public function hook() {
+		$slug = $this->slug;
+
+		tribe_notice(
+			$slug,
+			[ $this, 'notice' ],
+			[
+				'type'    => 'error',
+				'dismiss' => 1,
+				'wrap'    => 'p',
+			],
+			[ $this, 'should_display' ]
+		);
+	}
+
+	/**
+	 * Checks if we are in an TEC page or in main admin Dashboard.
+	 *
+	 * @since
+	 *
+	 * @return boolean
+	 */
+	public function should_display() {
+		global $pagenow;
+
+		if ( tribe( 'admin.helpers' )->is_screen() || 'index.php' === $pagenow ) {
+			return $this->is_rest_api_blocked() ? true : false;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Checks if our endpoints are accessible.
+	 *
+	 * @since
+	 *
+	 * @return boolean
+	 */
+	public function is_rest_api_blocked() {
+
+		$event_api = get_rest_url( null, '/tribe/events/v1/' );
+		$response  = wp_remote_get( $event_api );
+		if ( is_wp_error( $response ) || wp_remote_retrieve_response_code( $response ) !== 200 ) {
+			$this->blocked_endpoint = $event_api;
+			return true;
+		}
+
+		$views_api = get_rest_url( null, 'tribe/views/v2/' );
```

Comment:
These can be filtered and modified on the REST endpoint, we should use the methods there to get the URL we are testing.